model/src/pyrenew/model/rtinfectionsrenewalmodel.py

```diff
@@ -311,57 +313,41 @@ def sample(
 
         # Sampling initial infections
         I0, *_ = self.sample_I0(**kwargs)
-        I0_size = I0.size
         # Sampling from the latent process
-        latent_infections, *_ = self.sample_infections_latent(
+        post_seed_latent_infections, *_ = self.sample_infections_latent(
             Rt=Rt,
             gen_int=gen_int,
             I0=I0,
             **kwargs,
         )
 
-        if data_observed_infections is None:
-            (
-                observed_infections,
-                *_,
-            ) = self.sample_infection_obs_process(
-                observed_infections_mean=latent_infections,
-                data_observed_infections=data_observed_infections,
-                **kwargs,
-            )
-        else:
-            data_observed_infections = au.pad_x_to_match_y(
-                data_observed_infections,
-                latent_infections,
-                jnp.nan,
-                pad_direction="start",
-            )
+        if data_observed_infections is not None:
+            data_observed_infections = data_observed_infections[padding:]
 
-            (
-                observed_infections,
-                *_,
-            ) = self.sample_infection_obs_process(
-                observed_infections_mean=latent_infections[
-                    I0_size + padding :
-                ],
-                data_observed_infections=data_observed_infections[
-                    I0_size + padding :
-                ],
-                **kwargs,
-            )
+        observed_infections, *_ = self.sample_infection_obs_process(
+            observed_infections_mean=post_seed_latent_infections[padding:],
+            data_observed_infections=data_observed_infections,
+            **kwargs,
+        )
+
+        all_latent_infections = jnp.hstack([I0, post_seed_latent_infections])
+        npro.deterministic("latent_infections", all_latent_infections)
 
         observed_infections = au.pad_x_to_match_y(
             observed_infections,
-            latent_infections,
+            all_latent_infections,
```

Comment:
Alternate option: `all_latent_infections` &rarr; `total_latent_infections` 

Thoughts?

Reply:
I prefer `all` as it doesn't imply summation which `total` might.

Reply:
Okay, seems good to me.